src/js/components/Tabs/Tabs.js

```diff
@@ -93,37 +185,150 @@ const Tabs = forwardRef(
+                disabled={disableLeftArrow}
+                // removed from tabIndex, button is redundant for keyboard users
+                tabIndex={-1}
+                onClick={() => {
+                  let scrolledToIndex;
+                  for (let i = 0; i < tabRefs.length - 1; i += 1) {
+                    if (
+                      !isVisible(tabRefs[i].current) &&
+                      isVisible(tabRefs[i + 1].current)
+                    ) {
+                      if (scroll && scroll.step)
+                        i = Math.max(i - (scroll.step - 1), 0);
+                      scrolledToIndex = i;
+                      tabRefs[i].current.scrollIntoView({ behavior: 'smooth' });
+                      break;
+                    }
+                  }
+
+                  setDisableRightArrow(false);
+                  if (scrolledToIndex === 0) {
+                    // wait for scroll animation to finish
+                    const checkVisible = setInterval(() => {
+                      if (isVisible(tabRefs[0].current)) {
+                        setDisableLeftArrow(true);
+                        clearInterval(checkVisible);
+                      }
+                    }, 100);
+                    setTimeout(() => {
+                      clearInterval(checkVisible);
+                    }, 500);
+                  }
+                }}
+              />
+            )}
+            <StyledTabsHeader
+              ref={headerRef}
+              as={Box}
+              direction="row"
+              justify={justify}
+              alignSelf={alignControls}
+              flex={!!scroll}
+              wrap={!scroll}
+              overflow={scroll ? 'hidden' : 'visible'}
+              background={theme.tabs.header.background}
+              gap={theme.tabs.gap}
+              onBlur={() => setFocusIndex(activeIndex)}
+              {...tabsHeaderStyles}
+            >
+              {tabs}
+            </StyledTabsHeader>
+            {overflow && scroll && (
+              <Button
+                a11yTitle="Next Tab"
+                icon={<Next />}
+                disabled={disableRightArrow}
+                // removed from tabIndex, button is redundant for keyboard users
+                tabIndex={-1}
+                onClick={() => {
+                  let scrolledToIndex = 0;
+                  for (let i = tabRefs.length - 1; i > 0; i -= 1) {
+                    if (
+                      !isVisible(tabRefs[i].current) &&
+                      isVisible(tabRefs[i - 1].current)
+                    ) {
+                      if (scroll && scroll.step) {
+                        i = Math.min(i + (scroll.step - 1), tabRefs.length - 1);
+                      }
+                      scrolledToIndex = i;
+                      tabRefs[i].current.scrollIntoView({ behavior: 'smooth' });
+                      break;
+                    }
+                  }
+
+                  setDisableLeftArrow(false);
```

Comment:
Can we be sure we want to set it to `false`, would it be more reliable to `updateArrowStates`?

Reply:
I think it is safe to set this to false here because we know that the arrows will only show up if not all the tabs can fit on the screen. If we are able to press the right arrow we know we will move to the right by some amount and if we move to the right by any amount the left arrow should be enabled. The reason I didn't use `updateArrowStates` here is because we would need to wait for the animation to finish and I didn't want there to be a delay if we could avoid it.